metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -208,8 +208,7 @@ public QuorumController build() throws Exception {
                     "org.apache.kafka.controller.MockControllerMetrics").getConstructor().newInstance();
```

Comment:
Is this a potential issue? `"org.apache.kafka.controller.MockControllerMetrics"` is not in produce code also.

Reply:
I asked the same question and it's apparently intended. I think we should not use this pattern, but we can do that in a separate PR. This PR is required to fix some tests.

Reply:
It's not an issue because of the way it's used in the code.  But we will probably clean this up a little bit in a follow-on PR.